### ISSUE TITLE
Firefox menus don’t have rounded corners

### ISSUE THREAD

**Issue by matthewpaulthomas** [ubuntu/yaru#846]
yaru-theme-* 18.10.4, Ubuntu Cosmic

Menus have rounded corners in native GTK apps…

![menu pulldown in files](https://user-images.githubusercontent.com/19801137/45883863-2f45e580-bdaa-11e8-9a90-0d69ab948a90.png) ![context menu gtk](https://user-images.githubusercontent.com/19801137/45883862-2ead4f00-bdaa-11e8-88c2-38fd194c0690.png)

…and even in LibreOffice…

![menu in libreoffice](https://user-images.githubusercontent.com/19801137/45883833-1806f800-bdaa-11e8-93a8-f5e2220cb39e.png)

…but not in Firefox.

![menu pulldown in firefox](https://user-images.githubusercontent.com/19801137/45883891-408ef200-bdaa-11e8-9ab8-0016eb157420.png) ![menu context in firefox](https://user-images.githubusercontent.com/19801137/45883890-408ef200-bdaa-11e8-8ccb-19dfd48d3764.png)

This is odd because Firefox _does_ correctly include the tail at the top of dropdown menus.

Maybe this is not possible without code changes in Firefox: if so, it could be mentioned [in the wiki](https://github.com/ubuntu/yaru/wiki/%233-GTK-Theme-Design#menus).

[[Originally reported in the Yaru forum](https://discourse.ubuntu.com/t/call-for-participation-an-ubuntu-default-theme-lead-by-the-community/1545/1750).]

**Comment by Feichtmeier** [comment]
Yes Firefox has a bug with rounded edges. The border bleeds it of it and you end up with two borders.
As I said, Firefox does not properly suck up the gtk3 code.
See: https://discourse.ubuntu.com/t/firefox-theming/5920
(This was a long time ago and we even managed to apply the light blue to the text selection with some dirty CSS tricks which are wrong but we did it for the sake of consistency)

**Comment by Feichtmeier** [comment]
https://github.com/ubuntu/yaru/issues/841

https://github.com/ubuntu/yaru/pull/485

**Comment by Feichtmeier** [comment]
What's even worse is that the context menu can't have our drop shadow. :/

**Comment by clobrano** [comment]
Same thing happens with PopOS, rounded corner in libreoffice, squared and with different hover effect in Firefox.
Does anybody know the appropriate channel to report this? Bug tracker, IRC, other?
After reporting this, I believe we can close this issue

**Comment by matthewpaulthomas** [comment]
Coincidentally, Mozilla [bug 1489097 “[CSD] Enable round corners by default”](https://bugzilla.mozilla.org/show_bug.cgi?id=1489097) was fixed a week ago. So, I’d wait until the followup [bug 1493145](https://bugzilla.mozilla.org/show_bug.cgi?id=1493145) is fixed, then try a nightly build, or Firefox 64.

If that’s not the problem, “[Core > Widget: Gtk](https://bugzilla.mozilla.org/buglist.cgi?resolution=---&classification=Components&query_format=advanced&component=Widget%3A%20Gtk&product=Core)” is probably the right component.

**Comment by Feichtmeier** [comment]
That would be cool! It's fixed within a second in our code. Thanks for the info
To test if the round corner work you would need our css to be change. 
So it would be good to make it the other way round.
Tell us when it's merged, then I make a test PR for you

**Comment by clobrano** [comment]
I understand that Yaru is not the problem, but I think it doesn't harm us to use workarounds to fix wrong behaviors in important applications.

**Comment by clobrano** [comment]
> Coincidentally, Mozilla [bug 1489097 “[CSD] Enable round corners by default”](https://bugzilla.mozilla.org/show_bug.cgi?id=1489097) was fixed a week ago. So, I’d wait until the followup [bug 1493145](https://bugzilla.mozilla.org/show_bug.cgi?id=1493145) is fixed, then try a nightly build, or Firefox 64.
> 
> If that’s not the problem, “[Core > Widget: Gtk](https://bugzilla.mozilla.org/buglist.cgi?resolution=---&classification=Components&query_format=advanced&component=Widget%3A%20Gtk&product=Core)” is probably the right component.

Nightly (v65) does not seem to work as well

![image](https://user-images.githubusercontent.com/2883614/48700064-5cb4e100-ebeb-11e8-8ea1-1cf5e11f5163.png)

**Comment by Feichtmeier** [comment]
Different topic, but while working on the dark menus I've found a better solution for this glitch which leaves the shadow and doesnt cut it, which makes the context menu for firefox look a bit better
Just a reminder that we/I could add this

**Comment by Feichtmeier** [comment]
I just remembered that I already reported the problem with the wrong text selection colour
https://bugzilla.mozilla.org/show_bug.cgi?id=1461621
